servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -3454,4 +3476,61 @@ public Quota getQuota(String domainName) {
             return quotaCheck.getDomainQuota(con, domainName);
         }
     }
+
+    public void executePutRoleSystemMeta(ResourceContext ctx, String domainName, String roleName, RoleSystemMeta meta, String attribute, boolean deleteAllowed, String auditRef, String caller) {
+
+        // our exception handling code does the check for retry count
+        // and throws the exception it had received when the retry
+        // count reaches 0
+
+        for (int retryCount = defaultRetryCount; ; retryCount--) {
+
+            try (ObjectStoreConnection con = store.getConnection(false, true)) {
+
+                // first verify that auditing requirements are met
+
+                Domain domain = checkDomainAuditEnabled(con, domainName, auditRef, caller, getPrincipalName(ctx));
+
+                if (!domain.getAuditEnabled()) {
+                    throw ZMSUtils.requestError(caller + ": auditenabled flag not set for domain: " + domainName + " to add it on the role: " + roleName, caller);
+                }
+
+                Role rolefromdb = getRole(domainName, roleName, false, true);
```

Comment:
the last argument should be false as we don't need to expand any users

Reply:
fixed